MainRepository/Program.cs

```diff
@@ -1,7 +1,185 @@
+   }
+
+   /// <summary>Returns the random guess word</summary>
+   public static string SelectWord () {
+      string[] lines = File.ReadAllLines (@"c:\etc\puzzle-5.txt");
+      Random random = new ();
+      return lines[random.Next (lines.Length)];
+   }
+
+   /// <summary>Returns true if the given word is valid</summary>
+   public bool ValidWord () {
+      string givenString = string.Join ("", mList).ToUpper ();
+      string[] validWord = File.ReadAllLines (@"c:\etc\dict-5.txt");
+      return validWord.Contains (givenString);
+   }
+
+   /// <summary>Update the game-state based on the key the user pressed</summary>
+   public void UpdateGameState (string word) {
+      for (int i = 0; i < 6; i++) {
+         for (int j = 0; j < 5; j++) {
+            if (mWordle[i, j] == mCircle) {
+               mRow = i;
+               ConsoleKeyInfo key = Console.ReadKey (true);
+               if (key.Key is ConsoleKey.Backspace or ConsoleKey.LeftArrow) {
+                  if (j == 0) j--;
+                  else {
+                     mWordle[i, j - 1] = mCircle;
+                     mWordle[i, j] = mDot;
+                     j -= 2;
+                     mList.RemoveAt (mList.Count - 1);
+                  }
+               } else if (char.ToUpper (key.KeyChar) is >= 'A' and <= 'Z') {
+                  if (j < 4) mWordle[i, j + 1] = mCircle;
+                  char c = Convert.ToChar (key.KeyChar);
+                  mWordle[i, j] = char.ToUpper (c);
+                  if (mList.Count < 5) mList.Add (c);
+               }
+               if (i < 5 && j == 4) mWordle[i + 1, 0] = mCircle;
+               if (j == 4) {
+                  DisplayBoard (word);
+                  while (true) {
+                     var lkey = Console.ReadKey (true);
+                     if (lkey.Key == ConsoleKey.Enter && ValidWord ()) {
+                        Console.WriteLine (new string (' ', Console.WindowWidth));
+                        break;
+                     }
+                     if (lkey.Key == ConsoleKey.Enter && !ValidWord ()) Console.WriteLine ("Not a valid word");
+                     if (lkey.Key == ConsoleKey.Backspace) {
+                        mWordle[i, 4] = mCircle;
+                        mWordle[i + 1, 0] = mDot;
+                        j--;
+                        mList.RemoveAt (4);
+                        break;
+                     }
+                  }
+               }
+               DisplayBoard (word);
+               if (mList.Count == 5) {
+                  DisplayBoard (word, true);
+                  if (mGreenList.Count == 5) {
+                     mGameOver = true;
+                     mCount++;
+                     return;
+                  }
+                  if (!ValidWord ()) j--;
+                  else mList.Clear ();
+               }
+            }
+         }
+         mCount++;
+      }
+   }
+
+   /// <summary>Set the foreground color for the list</summary>
+   public char Color (char clr) {

```

Comment:
Naming of the char as clr is not good.
You are getting clr as input and returning the same. Why?
Name of the method is also not good. This should be called as SetForegroundClr (char ch)